### PR TITLE
Add tests for enum attributes in System.ComponentModel.Primitives

### PR DESCRIPTION
**Pull request by hughbe**
~I moved some code to ComponentModel.Primitives and saw that theres ~70% coverage. So here are some tests to bring it up to 100%.~

I've now split up this PR as it was v large. Together, they increase CC to 100%

### CODE REVIEW

File: src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs [hunk 1]
```diff
@@ -4,78 +4,72 @@
...
 using Xunit;
 
-namespace System.ComponentModel.Primitives.Tests
+namespace System.ComponentModel.Tests
 {
     public class EventHandlerListTests
     {
         [Fact]
-        public void AddHandler_Getter_RemoveHandler_Getter_Roundtrips()
+        public void AddHandle_RemoveHandle_Roundtrips()
```
Comment (stephentoub) [inline src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs]: Nit: AddHandle => AddHandler

File: src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs [hunk 2]
```diff
@@ -4,78 +4,72 @@
...
 
-            // Create two different delegate instances
-            Action a1 = () => Assert.True(false);
-            Action a2 = () => Assert.False(true);
-            Assert.NotSame(a1, a2);
+            Action action1 = () => Assert.True(false);
+            Action action2 = () => Assert.False(true);
 
-            // Neither entry in the list has a delegate
-            Assert.Null(list["key1"]);
```
Comment (stephentoub) [inline src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs]: Why did you remove this check?

File: src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs [hunk 3]
```diff
@@ -4,78 +4,72 @@
...
+            // Add the first delegate to the first entry.
+            list.AddHandler("key1", action1);
+            Assert.Same(action1, list["key1"]);
             Assert.Null(list["key2"]);
 
-            // Add the first delegate to the first entry
-            list.AddHandler("key1", a1);
-            Assert.Same(a1, list["key1"]);
-            Assert.Null(list["key2"]);
```
Comment (stephentoub) [inline src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs]: Why did you remove these?

File: src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs (hunk 2 above)
Comment (hughbe) [inline src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs]: This test moved to `Indexer_GetNoSuchKey_ReturnsNull`

File: src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs (hunk 3 above)
Comment (hughbe) [inline src/System.ComponentModel.Primitives/tests/EventHandlerListTests.cs]: I didn't. See L20-L22 in the new code:

```
// Add the first delegate to the first entry.
list.AddHandler("key1", action1);
Assert.Same(action1, list["key1"]);
Assert.Null(list["key2"]);
```

**Comment by hughbe** [comment]
Addressed PR feedback, thanks

**Comment by hughbe** [comment]
Good to merge?

**Comment by hughbe** [comment]
Sorry for the ping - is everyone happy with this PR? :)
